src/java.desktop/windows/classes/sun/awt/Win32GraphicsDevice.java

```diff
@@ -463,6 +463,11 @@ public synchronized void setDisplayMode(DisplayMode dm) {
             WWindowPeer peer = AWTAccessor.getComponentAccessor().getPeer(w);
             configDisplayMode(screen, peer, dm.getWidth(), dm.getHeight(),
                 dm.getBitDepth(), dm.getRefreshRate());
+            // configDisplayMode() changes the screen resolution (and potentially DPI) immediately,
+            // so we need to update the DPI scales here as well,
+            // otherwise getDefaultConfiguration().getBounds() will return incorrect results until
+            // WM_DISPLAYCHANGE or WM_DPICHANGED is received, which happens to be too late for our purposes
+            initScaleFactors();
```

Comment:
So WM_DISPLAYCHANGE message is received and processed, followed by WM_DPICHANGED, which updates the scaling factors used by Win32GraphicsConfig.getBounds() to the correct value of 100%."

causes Win32GraphicsDevice.displayChanged() to be called, which makes the update, but too late.

Is there anything else that method does which is being done too late ?

Do we still need the call to initScaleFactors() in Win32GraphicsDevice.displayChanged() ?

And can you reduce the line length of all these comments to <= 80 chars